### DISCUSSION TITLE
How can I render a text of markdown in nuxt v3

### DISCUSSION THREAD

**Discussion by mohammedhkmet** [community/community#129429] [Programming Help]
### Body

Make sure it is rendering on the server side for SEO reasons.

```vue
<script setup lang="ts">
const markdown = "# hello world"
</script>

<template>
  <LayoutContent>
    <!-- renders "markdown" here -->
  </LayoutContent>
</template>
```

I wanted to compile "markdown" into html and render it inside a LayoutContent component.
In addition, I want it to support code highlight.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by yhemn** [comment]
**Step 1**: Install the packages `markdown-it` and `highlight.js` by running the following command in your terminal:
```bash
npm install -D markdown-it highlight.js
```

**Step 2**: Update your `nuxt.config.ts` file by enabling support for `.server.vue` extensions, which enables Server-Side Rendering. To achieve this, add the experimental property with the value `componentIslands: true`.
```typescript
export default defineNuxtConfig({
  // ...
  experimental: {
    componentIslands: true,
  },
})
```

**Step 3**: Create a new component named `HighlightedMarkdown.server.vue` and save it in the `components` directory, then set the content of the component to the following:
```vue
<script setup lang="ts">
import markdownit from 'markdown-it'
import hljs from 'highlight.js'

interface PropsType {
  markdown: string
}

const { markdown } = defineProps<PropsType>()

const md = markdownit({
  highlight(str, lang) {
    if (lang && hljs.getLanguage(lang)) {
      try {
        return hljs.highlight(str, { language: lang }).value
      }
      catch (__) {}
    }

    return '' // use external default escaping
  },
})
const html = md.render(markdown)
</script>

<template>
  <div class="prose prose-lg" v-html="html" />
</template>
```

**Step 4**: To enable code highlighting, you can install a theme CSS file. You can find more information [here](https://highlightjs.org/examples).

If you're using tailwindcss in your application, you'll need to include the [@tailwindcss/typography](https://tailwindcss-typography.vercel.app/) plugin. To add the plugin, follow these simple steps:

1. Run the following command to install the plugin:
```bash
npm install -D @tailwindcss/typography
```

2. In your `tailwind.config.ts` file, import the plugin:
```typescript
import tailwindTypography from '@tailwindcss/typography'
```

3. Add the plugin to the `plugins` inside your `tailwind.config.ts` file:
```typescript
plugins: [tailwindTypography()]
```

Now that all the necessary steps have been completed, you can easily add the `HighlightedMarkdown` component wherever you need it in your project.

```vue
<script setup lang="ts">
const markdown = `
# Heading 1
## Heading 2
### Heading 3
#### Heading 4

\`\`\`javascript
function myFunction () {
  console.log("Hello World")
}
\`\`\`
`
</script>

<template>
  <LayoutContent>
    <HighlightedMarkdown :markdown="markdown" />
  </LayoutContent>
</template>
```

**Comment by ebndev** [comment]
Hi @mohammedhkmet, thanks for being a part of the GitHub Community!

While we appreciate you wanting to contribute to Discussions, please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
